GUI/Types/Renderer/ScenePostProcessVolume.cs

```diff
@@ -0,0 +1,257 @@
+            PostProcessingResource = resource;
+
+            var tonemapParams = PostProcessingResource.GetTonemapParams();
+
+            if (tonemapParams != null)
+            {
+                HasTonemap = true;
+                PostProcessTonemapSettings = new TonemapSettings(tonemapParams);
+            }
+
+            // HLA has slightly different behavior. TODO
+            IsPostHLA = PostProcessingResource.Data.ContainsKey("m_bHasLocalContrastParams");
+
+            // Create color correction texture from raw data
+            if (resource.HasColorCorrection())
+            {
+                var dimensions = resource.GetColorCorrectionLUTDimension();
+                var data = resource.GetRAWData();
+                var bytesPerPixel = 4; // currently a constant 4 byte per pixel, one per channel
+
+                ColorCorrectionLUT = new RenderTexture(TextureTarget.Texture3D, dimensions, dimensions, dimensions, 1);
+                ColorCorrectionLUT.SetWrapMode(TextureWrapMode.ClampToEdge);
+                ColorCorrectionLUT.SetFiltering(TextureMinFilter.Linear, TextureMagFilter.Linear);
+                GL.TextureStorage3D(ColorCorrectionLUT.Handle, 1, SizedInternalFormat.Rgba8, dimensions, dimensions, dimensions);
+                // COLOR CORRECTION DEBUG
+#if true
+                Log.Info(nameof(ScenePostProcessVolume), "CREATING COLOR CORRECTION TEXTURE");
+                Log.Info(nameof(ScenePostProcessVolume), $"Dimensions: {dimensions} x {dimensions} x {dimensions}");
+                Log.Info(nameof(ScenePostProcessVolume), $"Total Size in Bytes: {data.Length}");
+                Log.Info(nameof(ScenePostProcessVolume), $"Number of Blocks: {data.Length / bytesPerPixel}");
+                Log.Info(nameof(ScenePostProcessVolume), $"LUT Texture Handle: {ColorCorrectionLUT.Handle}");
+
+                Log.Info(nameof(ScenePostProcessVolume), "PRINTING FIRST 32X32 BLOCK");
+                for (var i = 0; i < ((dimensions * dimensions) / bytesPerPixel); i++)
+                {
+                    var pixelValues = new string("");
+                    for (var k = 0; k < bytesPerPixel; k++)
+                    {
+                        if (k == 0)
+                        {
+                            pixelValues = $"{data[i * 4 + k] / 255.0f} ";
+                        }
+                        else
+                        {
+                            pixelValues = pixelValues.Insert(pixelValues.Length - 1, $"{data[i * 4 + k] / 255.0f} ");
+                        }
+                    }
+                    Log.Info(nameof(ScenePostProcessVolume), pixelValues);
+                }
+#endif
+
+                var buffer = ArrayPool<byte>.Shared.Rent(dimensions * dimensions * dimensions * bytesPerPixel);
+
+                try
+                {
+                    GL.TextureSubImage3D(ColorCorrectionLUT.Handle, 0, 0, 0, 0, dimensions, dimensions, dimensions, PixelFormat.Rgba, PixelType.UnsignedByte, buffer);
```

Comment:
A buffer is being rented, but nothing is written to it, so this writes garbage data to the texture (im surprised this worked at all on any map).

Did you mean to write `data` here without buffer?

Also keep in mind ArrayPool can return bigger sizes than requested.

Reply:
> im surprised this worked at all on any map

I'm guessing it rented same array as the one rented by binarykv3.